### PR TITLE
Cleaner Search Page and Results from DB

### PR DESCRIPTION
**Pull request by christinajiyunlee**
This PR creates a new cleaner search page that displays all cleaners from DB. There is a filter UI available that will implement filtering logic in the next sprint.

### CODE REVIEW

File: server/node_modules/buffer-writer/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+language: node_js
```
Comment (gurleen-sanghera) [inline server/node_modules/buffer-writer/.travis.yml]: @kvptkr @nicolasabousawan should these node_modules files all be committed here?

File: server/searchPage.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const { Pool, Client } = require('pg')
+
+const pool = new Pool({// !!! CHANGE
+	// user: process.env.user,
+	// host: process.env.host,
+ // 	database: process.env.database,
+ //  	password: process.env.password,
+ //  	port: process.env.port,
+ 	user: 'master',
```
Comment (gurleen-sanghera) [inline server/searchPage.js]: need to comment out the credentials here!!

File: ui/src/Components/SearchCandidates.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	// 	this.setState({search: data, loading:false})
+	// }
+	componentDidMount() {
+		fetch('https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/search')
+		  .then(response => response.json())
+		  .then(data => this.setState({ search: data, loading: false }));
+	}
+
+	/** Event Handlers for search fields
+		// TODO: Uncomment so state is updated, triggering a rerender of results
```
Comment (gurleen-sanghera) [inline ui/src/Components/SearchCandidates.jsx:41]: Is this a to-do for future sprints?

File: ui/src/Components/SearchCandidates.jsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	render(){
+		console.log("RENDERING");
+		// Add results row by row
+		if (this.state.search) {
+			const obj = this.state.search.result; // instance of cleaner data queried earlier
+			var objLen = Object.keys(obj).length;
+			var items = [];
+
+			// Builds table with JSON object items passed in
+			items = []; // rebuild
```
Comment (gurleen-sanghera) [inline ui/src/Components/SearchCandidates.jsx]: I'm not familiar with what this does, so just flagging it as it isn't something I have seen (declaring the variable items, then setting it to an empty array again). If this is intentional, then ignore :)

**Review by gurleen-sanghera — CHANGES_REQUESTED** [review]
Yay looks good! Make sure to remove the DB connection stuff + left some other comments. I'd like to wait for someone else to also look over before approving.

File: ui/src/Components/SearchCandidates.jsx (hunk 3 above)
Comment (christinajiyunlee) [reply ui/src/Components/SearchCandidates.jsx:41]: yes, just a starting point for the filter function!

File: ui/src/Components/SearchCandidates.jsx (hunk 4 above)
Comment (christinajiyunlee) [reply ui/src/Components/SearchCandidates.jsx]: Between the two items = [] there used to be a function declaration but now this doesn't make sense, I'll remove the second items declaration!

File: ui/src/Components/search-box/search-box.component.jsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import React from "react"
+import "./search-box.styles.css"
+export const SearchBox = ({ placeholder, handleChange }) => (
+  <input type="search" placeholder={placeholder} onChange={handleChange} />
+)
```
Comment (kvptkr) [inline ui/src/Components/search-box/search-box.component.jsx]: nit: matt loooves his new lines at the end of files :P

File: ui/src/Components/search-box/search-box.styles.css [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  -webkit-transition: width 0.4s ease-in-out;
+  transition: width 0.4s ease-in-out;
+  width: 130px;
+  border: 3px solid aquamarine;
+  padding: 8px;
+}
+/* When input field gets focus, change width*/
+input:focus {
+  width: 70%;
+}
```
Comment (kvptkr) [inline ui/src/Components/search-box/search-box.styles.css]: newline at the end of file

File: ui/src/Components/SearchCandidates.jsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+					console.log(searchName);
+					finalResults = fullName.includes(String.prototype.toLowerCase.call(searchName));
+				}
+				return finalResults	
+
+			var filteredResults = obj.filter(applyFilters);
+			displayResults(filteredResults);
+			*/
+
+
```
Comment (kvptkr) [inline ui/src/Components/SearchCandidates.jsx:97]: 👾

File: ui/src/Components/SearchCandidates.jsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+				// !!! Link row to profile eventually
+				items.push(<tr>
+							<th scope="row">{i+1}</th>
+							<td ><a href={"/profiles/"+obj[i].profile_id}>{obj[i].first_name} {obj[i].last_name}</a></td>
+							<td>{obj[i].email}</td>
+							<td>{obj[i].contact_num}</td>
+						</tr>)
+			}
+
+
```
Comment (kvptkr) [inline ui/src/Components/SearchCandidates.jsx:75]: 👾 👾

File: ui/src/Components/SearchCandidates.jsx [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+	// {
+	// 	const url = `https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/search`;
+	// 	const resp = await fetch(url);
+	// 	const data = await resp.json();
+	// 	this.setState({search: data, loading:false})
+	// }
+	componentDidMount() {
+		fetch('https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/search')
+		  .then(response => response.json())
+		  .then(data => this.setState({ search: data, loading: false }));
```
Comment (kvptkr) [inline ui/src/Components/SearchCandidates.jsx]: I think that we should have a `.catch` here, just in case it errors out

File: ui/src/App.js [hunk 10]
```diff
@@ -22,10 +23,14 @@ function App() {
...
+        <Route exact path="/search">
+            <SearchCandidates />
+        </Route>
       </Switch>
       </div>
     </Router>
   );
 }
 
+
```
Comment (kvptkr) [inline ui/src/App.js:35]: 👾

File: ui/src/Components/SearchCandidates.jsx [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+			loading: true,
+			search: null,
+			props: props,
+			searchName: null,
+			searchEmail: null,
+			searchNumber: null,
+		};
+	}
+
+
```
Comment (kvptkr) [inline ui/src/Components/SearchCandidates.jsx:20]: 👾

**Review by kvptkr — CHANGES_REQUESTED** [review]
All your actual changes lgtm p much, had to really nit. The only thing is we don't want to commit anything in the node_modules folder, we should add `node_modules` to the .gitignore and then `git rm -rf node_modules` from your branch.

File: server/node_modules/buffer-writer/.travis.yml (hunk 1 above)
Comment (christinajiyunlee) [reply server/node_modules/buffer-writer/.travis.yml]: ran git rm and verified that node_modules is in the .gitignore file

File: ui/src/Components/search-box/search-box.component.jsx (hunk 5 above)
Comment (christinajiyunlee) [reply ui/src/Components/search-box/search-box.component.jsx]: lul I removed 'em

File: ui/src/Components/SearchCandidates.jsx (hunk 9 above)
Comment (christinajiyunlee) [reply ui/src/Components/SearchCandidates.jsx]: added .catch!

**Review by kvptkr — APPROVED** [review]

**Pull request merged**